mesonbuild/compilers/compilers.py

```diff
@@ -924,7 +924,15 @@ def get_gcc_soname_args(gcc_type, prefix, shlib_name, suffix, path, soversion, i
         if soversion is not None:
             install_name += '.' + soversion
         install_name += '.dylib'
-        return ['-install_name', os.path.join('@rpath', install_name)]
+        args = ['-install_name', os.path.join('@rpath', install_name)]
+        if version and len(version.split('.')) == 3:
+            splitted = version.split('.')
+            major = int(splitted[0])
+            minor = int(splitted[1])
+            revision = int(splitted[2])
+            args += ['-compatibility_version', '%d' % (major + minor + 1)]
```

Comment:
This seems a bit baffling. It would mean that versions 1.2.3 and 2.1.3 would get the same compatibility version. Is that really what should happen? Is there a link to Apple's documentation that says what this should do? I tried looking some time ago but the docs I found were most unhelpful.